src/mscorlib/shared/System/Globalization/DateTimeFormat.cs

```diff
@@ -966,8 +969,32 @@ internal static String Format(DateTime dateTime, String format, DateTimeFormatIn
             return Format(dateTime, format, dtfi, NullOffset);
         }
 
+        internal static string Format(DateTime dateTime, String format, DateTimeFormatInfo dtfi, TimeSpan offset) =>
+            StringBuilderCache.GetStringAndRelease(FormatStringBuilder(dateTime, format, dtfi, offset));
 
-        internal static String Format(DateTime dateTime, String format, DateTimeFormatInfo dtfi, TimeSpan offset)
+        internal static bool TryFormat(DateTime dateTime, Span<char> destination, out int charsWritten, string format, DateTimeFormatInfo dtfi) =>
+            TryFormat(dateTime, destination, out charsWritten, format, dtfi, NullOffset);
+
+        internal static bool TryFormat(DateTime dateTime, Span<char> destination, out int charsWritten, string format, DateTimeFormatInfo dtfi, TimeSpan offset)
+        {
+            StringBuilder sb = FormatStringBuilder(dateTime, format, dtfi, offset);
+
+            bool success = sb.Length <= destination.Length;
+            if (success)
+            {
+                sb.CopyTo(0, destination, sb.Length);
+                charsWritten = sb.Length;
+            }
+            else
+            {
+                charsWritten = 0;
+            }
+
+            StringBuilderCache.Release(sb);
```

Comment:
>StringBuilderCache.Release(sb) [](start = 12, length = 30)

did we release this here and not in FormatStringBuilder because of perf? my concern is anyone calling FormatStringBuilder has to know should release the SB from the cache. how much perf we are gaining here?

Reply:
> did we release this here and not in FormatStringBuilder because of perf?

No, because it's used by two different callers that need the StringBuilder to do different things with it: one calls ToString to get the results as a string, the other calls CopyTo to get the results into a destination span.